Bunkum.Protocols.Http/Socket/SocketHttpListener.cs

```diff
@@ -75,8 +81,53 @@ public override void StartListening()
         }
     }
 
-    private ListenerContext ReadRequestIntoContext(System.Net.Sockets.Socket client, Stream stream)
+    private ListenerContext ReadRequestIntoContext(System.Net.Sockets.Socket client, Stream rawStream)
     {
+        Stream stream = rawStream;
+        SslStream? sslStream = null;
+        if (this._cert != null)
+        {
+            sslStream = new SslStream(rawStream);
+
+            SslServerAuthenticationOptions authOptions = new()
+            {
+                //TODO: move this into the config
+                EnabledSslProtocols = SslProtocols.Tls12 | SslProtocols.Tls11,
```

Comment:
Why not TLS1.3 by default?

Reply:
if you enable TLS1.3 + the old ps3 ciphers we have enabled, it causes openssl to error on init, they seem to conflict with eachother for some reason, im guessing its something weird the C# stdlib is doing when setting up openssl

Reply:
Hmmm.. I think that should be okay for now.

I don't want people to really use HTTPS in Bunkum anyways unless they have to like our situation; the reverse proxy should be responsible for that. So I'm fine with it being a little weird for now.